django/db/models/query_utils.py

```diff
@@ -126,8 +125,9 @@ def __get__(self, instance, cls=None):
         Retrieves and caches the value from the datastore on the first lookup.
         Returns the cached value.
         """
-        non_deferred_model = instance._meta.proxy_for_model
-        opts = non_deferred_model._meta
+        if instance is None:
+            return
```

Comment:
I removed the assert in https://github.com/django/django/pull/6491. The if block is needed as `instance` sometimes is `None`.